bgpd/bgp_filter.c

```diff
@@ -634,6 +643,18 @@ static int config_write_as_list(struct vty *vty)
 	return write;
 }
 
+int config_bgp_aspath_validate(const char *regstr)
+{
+	size_t i;
+	char valid_chars[] = "1234567890_(^|[,{}() ]|$)";
+
+	for (i = 0; i < strlen(regstr); i++)
+		if (strchr(valid_chars, regstr[i]) == NULL)
```

Comment:
You can use `strspn()` here. (`strspn(regstr, valid_chars) - regstr == strlen(regstr)`)